plugins/org.eclipse.n4js/src/org/eclipse/n4js/projectModel/names/N4JSProjectName.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2019 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.projectModel.names;
+
+import org.eclipse.n4js.utils.ProjectDescriptionUtils;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * A project name of the shape `(@scope/)?name`.
+ */
+public class N4JSProjectName implements Comparable<N4JSProjectName> {
+
+	private final String name;
+
+	public N4JSProjectName(String name) {
+		this.name = Preconditions.checkNotNull(name);
+		Preconditions.checkArgument(name.indexOf(':') < 0, name);
```

Comment:
A similar check in `EclipseProjectName` could be introduced:
`Preconditions.checkArgument(name.indexOf('/') < 0, name);`